src/commands/math.ts

```diff
@@ -42,131 +47,147 @@ var MathElement = P(Node, function(_, super_) {
   };
   // Remove nodes that are more than `cutoff`
   // blocks deep from this node.
-  _.removeNodesDeeperThan = function (cutoff) {
+  removeNodesDeeperThan (cutoff:number) {
     var depth = 0;
-    var queue = [[this, depth]];
-    var current;
+    var queue:[[MQNode, number]] = [[this, depth]];
```

Comment:
I believe this type for the queue should be declared an array of tuples, rather than an tuple of tuples.

Tried learning why this code even works i.e. pushing and shifting from a tuple. Tuples are described in many places as “fixed length ordered collections”. My understanding is that this type definition should be declaring the outer tuple only ever has a single element. Apparently you can also treat a tuple like an array for methods like push/pop/shift and iterating over it, which makes this work but I assume it is discouraged as trying to reference any element past the 0th, with something like queue[1] will fail.

https://dev.to/johan_du_toit/typescript-tuples-are-just-fixed-length-arrays-449n

People have proposed fixing this in the language, but it would be a breaking change. Rather annoyingly people also say all over the place “tuples are just arrays”, which doesn’t make any sense to me. It’s a different type because it existing as a separate type is useful, and it would be nice if it just had consistent behavior.
https://stackoverflow.com/a/64070080/1387961
https://github.com/microsoft/TypeScript/issues/6325

Reply:
Tracked as https://github.com/mathquill/mathquill/issues/982